test/behaviour/concept/thing/entity/EntitySteps.jl

```diff
@@ -1,85 +1,216 @@
-#     }
-# 
-#     @Then("entity\\( ?{type_label} ?) get instances contain: {var}")
-#     public void entity_type_get_instances_contain(String typeLabel, String var) {
-#         assertTrue(tx().concepts().getEntityType(typeLabel).asRemote(tx()).getInstances().anyMatch(i -> i.equals(get(var))));
-#     }
-# 
-#     @Then("entity\\( ?{type_label} ?) get instances is empty")
-#     public void entity_type_get_instances_is_empty(String typeLabel) {
-#         assertEquals(0, tx().concepts().getEntityType(typeLabel).asRemote(tx()).getInstances().count());
-#     }
-# }
+using Base: concatenate_setindex!, @nexprs
+
+@given("put attribute type: username, with value type: string") do context
+    _put_attribute_to_db(context, "username", g.Proto.AttributeType_ValueType.STRING)
+end
+
+@given("entity(person) set owns key type: username") do context
+    g.con.entity_set_owns("person", "username", context[:transaction], true)
+end
+
+# Scenario: Entity can be created
+@when("\$a = entity(person) create new instance with key(username): alice") do context
+
+    username_type = g.get(context[:cm], AttributeType, "username")
+    username_alice = g.put(g.as_remote(username_type, context[:transaction]), "alice")
+
+    ent_type = g.get(context[:cm], EntityType, "person")
+    ent_instance = g.create(g.as_remote(ent_type, context[:transaction]))
+
+    g.set_has(context[:transaction], ent_instance, username_alice)
+
+    context[:a] = ent_instance
+end
+
+@then("entity \$a is null: false") do context
+    @expect context[:a] !== nothing
+end
+
+@then("entity \$a has type: person") do context
+    erg = context[:a]
+    @expect erg.type.label.name == "person"
+end
+
+@then("entity(person) get instances contain: \$a") do context
+    attr = g.get(context[:cm], EntityType, "person")
+    res = g.get_instances(g.as_remote(attr, context[:transaction]))
+    @expect in(context[:a], res)
+end
+
+@when("\$a = entity(person) get instance with key(username): alice") do context
+    attr_type = g.get(context[:cm], AttributeType, "username")
+    res_attr = g.get(g.as_remote(attr_type, context[:transaction]), "alice")
+    context[:a] = g.get_owners(context[:transaction], res_attr, nothing)[1]
+end
+
+# Scenario: Entity cannot be created when it misses a key
+@when("\$a = entity(person) create new instance") do context
+    ent_type = g.get(context[:cm], EntityType, "person")
+    context[:a] = g.create(g.as_remote(ent_type, context[:transaction]))
+end
+
+# Scenario: Entity can be deleted
+@when("delete entity: \$a") do context
+    g.delete(g.as_remote(context[:a], context[:transaction]))
+end
+
+@then("entity \$a is deleted: true") do context
+    res = g.is_deleted(g.as_remote(context[:a], context[:transaction]))
+    @expect res === true
+end
+
+@then("entity(person) get instances is empty") do context
+    attr = g.get(context[:cm], EntityType, "person")
+    res_ent = g.get_instances(g.as_remote(attr, context[:transaction]))
+    @expect isempty(res_ent) === true
```

Comment:
When the function clearly returns bool, there is no need to compare with `true`. Same comment for a few other cases below.
```suggestion
    @expect isempty(res_ent)
```

Reply:
For today it isn't possible to formulate only a variable which contains a Bool in the expect section. A issue is filed and a PR to reach the goal is submitted.

Reply:
Erik has merged the change what made it possible to fulfill your suggestion.